src/modules/frequency/application/frequency.service.ts

```diff
@@ -1,26 +1,258 @@
+  @Inject(FREQUENCY_QUERIES_TOKEN)
+  private readonly frequencyQueryService: IFrequencyQueries;
+  @Inject(FREQUENCY_REPOSITORY_TOKEN)
+  private readonly frequencyRepository: IFrequencyRepository;
+
+  public async getUserClasses(userId: number): Promise<UserClassesResponseDTO> {
+    const userClasses = await this.frequencyQueryService.getMyClasses(userId);
+    userClasses.push({
+      classId: null,
+      className: "Geral",
+      classState: "ATIVA",
+      levelName: null,
+      isGeral: true,
+      activity: {
+        activityId: null,
+        activityName: "Atividade geral",
+      },
+    });
+    return {
+      classes: userClasses,
+    };
+  }
+  public async getGeneralAttendance(
+    date: Date,
+  ): Promise<GeneralAttendanceResponseDTO> {
+    const studentList =
+      await this.frequencyQueryService.getGeneralAttendance(date);
+    return {
+      date: date.toISOString().split("T")[0],
+      studentList: studentList,
+    };
+  }
+  public async updateGeneralAttendance(
+    data: UpdateGeneralAttendanceRequestDTO,
+  ): Promise<boolean> {
+    const date = data.date;
+    const isConsistent = await this.isGeneralAttendanceConsistent(
+      data["studentList"],
+      date,
+    );
+    if (!isConsistent.isValid) {
+      throw new BadRequestException(
+        `Attendance permission conflict: The generalAttendanceAllowed values in your request don't match our system records. Problem with studentId: ${isConsistent.problematicId}`,
+      );
+    }
+    const validItems: UpdateGeneralAttendanceItemDTO[] = data[
+      "studentList"
+    ].filter((item) => item.generalAttendanceAllowed);
+    const present: Frequency[] = [];
+    const notPresent: Frequency[] = [];
+    const domainItems: Frequency[] = validItems.map(
+      (item) =>
+        new Frequency({
+          id: null,
+          studentId: item.studentId,
+          classId: null,
+          date: date,
+          status: item.status,
+          notes: null,
+        }),
+    );
+    for (let index = 0; index < domainItems.length; index++) {
+      const element = domainItems[index];
+      if (element.getStatus() === FrequencyStatus.PRESENTE) {
+        present.push(element);
+      } else if (element.getStatus() === FrequencyStatus.AUSENTE) {
+        notPresent.push(element);
+      }
+    }
+    if (notPresent.length) {
+      await this.frequencyRepository.deleteManyByStudentAndClassAndDate(
+        notPresent,
+      );
+    }
+    for (let index = 0; index < present.length; index++) {
+      await this.frequencyRepository.upsert(present[index]);
+    }
+    return true;
+  }
+  private async isGeneralAttendanceConsistent(
+    frequencyList: UpdateGeneralAttendanceItemDTO[],
+    date: Date,
+  ): Promise<{isValid:boolean, problematicId: number | null;}> {
+    const attendanceListFromDb =
+      await this.frequencyQueryService.getGeneralAttendance(date);
+    const stdntIdVsStatusMap: Map<number, boolean> = new Map();
```

Comment:
Como comentei em uma review anterior, não temos definições sobre abreviações, tampouco tenho alguma opinião muito forte contra ou a favor. Apenas comento caso julgue necessário padronizar para algum dos lados.
```suggestion
    const studentIdVsStatusMap: Map<number, boolean> = new Map<number, boolean>();
```

Reply:
Vou alterar direto no código.